Server/src/server/dataCenter/DataCenter.java

```diff
@@ -98,65 +98,71 @@ public String getClientName(String username) {
         return accounts.get(account);
     }
 
-    public void register(Message message) throws LogicException {
-        if (message.getAccountFields().getUsername() == null || message.getAccountFields().getUsername().length() < 2
-                || getAccount(message.getAccountFields().getUsername()) != null) {
+    private void login(Account account, String client) throws LogicException {
+        //TODO: send commented out parts async
+        if (!ClientPortal.getInstance().hasThisClient(client)) {
+            throw new LogicException("Client Wasn't Added!");
+        } else if (account == null) {
+            throw new ClientException("Username Not Found!");
+        } else if (accounts.get(account) != null) {
+            throw new ClientException("Selected Username Is Online!");
+        } else if (clients.get(client) != null) {
+            throw new ClientException("Your Client Has Logged In Before!");
+        } else {
+            accounts.replace(account, client);
+            clients.replace(client, account);
+            GameServer.addToSendingMessages(Message.makeAccountCopyMessage(client, account));
+            GameServer.serverPrint(client + " Is Logged In");
+        }
+    }
+
+    private void register(String username, String client) throws LogicException {
+        if (username == null || username.length() < 2
+                || getAccount(username) != null) {
```

Comment:
When people register, is there are database somewhere? 

If so, do we not also need to check in name has not been taken?

Reply:
Yes, and it is being checked both in this repo and the microservice doing the actual registration.